### PR TITLE
feat: add option to retry or cancel messages that failed to send [AR-2188]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2188" title="AR-2188" target="_blank">StoryAR-2188</a>  Retry/cancel when message fails to send
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Right now the user doesn’t have the option to retry or cancel messages that failed to send.

### Solutions

For retrying, use newly created use case.
For canceling, use existing logic to delete the message.

### Dependencies (Optional)

Needs releases with:

- https://github.com/wireapp/kalium/pull/1770

#### How to Test

Send a message when being offline and cancel or retry it.

### Attachments (Optional)

https://github.com/wireapp/wire-android-reloaded/assets/30429749/33f67078-b035-4ca6-b53a-c56b4f8c5bb4

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1830/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1830/3/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt [hunk 1]
```diff
@@ -98,6 +99,7 @@ class MessageComposerViewModel @Inject constructor(
     private val sendAssetMessage: ScheduleNewAssetMessageUseCase,
     private val sendTextMessage: SendTextMessageUseCase,
     private val sendEditTextMessage: SendEditTextMessageUseCase,
+    private val retryFailedMessageUseCase: RetryFailedMessageUseCase,
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt]: We don't need to add `useCase` suffix to the name

**Review by ohassine — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1830/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6663755)(6663755)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6663755)(6663755)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItemComponents.kt [hunk 2]
```diff
@@ -128,3 +138,15 @@ internal fun Modifier.customizeMessageBackground(
         this
     }
 }
+
+@Preview
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItemComponents.kt]:
```suggestion
@PreviewMultipleThemes
```

Maybe this one? :D

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItemComponents.kt [hunk 3]
```diff
@@ -128,3 +138,15 @@ internal fun Modifier.customizeMessageBackground(
...
+@Composable
+fun PreviewMessageSendFailureWarning() {
+    MessageSendFailureWarning(MessageStatus.SendFailure, {}, {})
+}
+
+@Preview
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItemComponents.kt]:
```suggestion
@PreviewMultipleThemes
```

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageTypesPreview.kt [hunk 4]
```diff
@@ -67,6 +68,33 @@ fun PreviewMessage() {
     )
 }
 
+@Preview(showBackground = true)
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageTypesPreview.kt]:
```suggestion
@PreviewMultipleThemes
```

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Looks great 👨‍🍳🤌🏻

I'll keep it as `Request changes` to wait for @ohassine's suggestion and mine to use `@PreviewMultipleThemes`.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1830/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6666017)(6666017)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6666017)(6666017)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1830/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6666883)(6666883)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6666883)(6666883)-staging-compat.apk)

**Review by vitorhugods — APPROVED** [review]
🧑🏻‍🍳🤌🏻

**Pull request merged**